obspy/io/sac/util.py

```diff
@@ -268,104 +284,114 @@ def obspy_to_sac_header(stats, keep_sac_header=True):
+        values introduced.  It will be an iztype 9 ("ib") file, with small
+        reference time adjustments for micro/milliseconds precision issues.
+        SAC headers nvhdr, level, lovrok, and iftype are always produced.
     :type keep_sac_header: bool
 
     """
-    # XXX: forces the user to keep either all or nothing of the old SAC header
-    # (almost).  e.g. updates to stats.channel are ignored for stats.sac.kcmpnm
-    # if you want to keep the old iztype & nz times with keep_sac_header=True.
     header = {}
     oldsac = stats.get('sac', {})
 
+    header['npts'] = stats['npts']
+    header['delta'] = stats['delta']
+
     if keep_sac_header and oldsac:
         # start with the old header, and only update a minimal set of headers
-        # values from stats: npts, b, e, delta
         header.update(oldsac)
 
-        header['npts'] = stats['npts']
-        header['delta'] = stats['delta']
-
-        # XXX: if you don't know "b", you don't know the difference btwn
-        #   the old and new 1st sample times, and b/bshift will be wrong.
-        #   ObsPy compatible behavior is to procede as though iztype is 'ib',
-        #   the starttime is the reftime, and b is 0.0?
-
+        # try to set "b" and "e"
+        # NOTE: if you don't know the old absolute first sample time, you don't
+        # know the difference btwn the old SAC 1st sample time and the current
+        # stats.starttime (in the case of Trace merging or trimming). If the
+        # old iztype was 9, knowing this shift is required in order to keep SAC
+        # relative time headers valid.
+        # XXX: what about synthetic data, sac funcgen files?
         try:
-            # if the first sample time in stats is different than the one in
-            # the old header (e.g. from trimming), we need to know that shift.
-            # The treatment will depend on whether the old header had a valid
-            # reftime.
             reftime = get_sac_reftime(header)
-            # reftme + b is the old first sample time
+            # reftime + b is the old first sample time
             b = stats['starttime'] - reftime
             # NOTE: if b or e is null, it will become set here.
-            # TODO: I want to uncomment this eventually.
             header['b'] = b
             header['e'] = b + (stats['endtime'] - stats['starttime'])
         except SacHeaderTimeError:
-            # can't determine absolute time shift.
-            # assume that the old and new 1st sample times are the same
+            # can't determine reftime or absolute time shift.
             msg = "Old header has invalid reftime."
             warnings.warn(msg)
+
+            # If no relative headers are set and the iztype was 9, we're OK to
+            # use stats.starttime as the reftime, and set b, e
+            # ObsPy issue 1204
+            # TODO: consolidate relative-time header list in header.py
+            relhdrs = ['t'+str(i) for i in range(10)] + ['a', 'f']
```

Comment:
Please add spaces around the `+`. I'm not sure why `flake8` does not catch some things.